package/cloudshell/iac/terraform/terraform_shell.py

```diff
@@ -1,98 +1,149 @@
 
-    @staticmethod
-    def _using_remote_state(shell_helper) -> bool:
+    def _using_remote_state(self, shell_helper) -> bool:
         return bool(shell_helper.attr_handler.get_attribute(ATTRIBUTE_NAMES.REMOTE_STATE_PROVIDER))
 
-    @staticmethod
-    def _does_working_dir_exists(directory: str) -> bool:
-        return directory and os.path.isdir(directory)
-
-    @staticmethod
-    def _handle_error_output(shell_helper: ShellHelperObject, err_msg: str):
-        shell_helper.sandbox_messages.write_error_message(err_msg)
-        raise Exception(err_msg)
+    def _delete_local_temp_dir(self, sandbox_data_handler, tf_working_dir):
+        tf_path = Path(tf_working_dir)
+        tmp_folder_found = False
+        while not tmp_folder_found:
+            objects_in_folder = os.listdir(tf_path.parent.absolute())
+            if len(objects_in_folder) == 2:
+                if 'REPO' in objects_in_folder and 'repo.zip' in objects_in_folder:
+                    tmp_folder_found = True
+            tf_path = Path(tf_path.parent.absolute())
+        tf_path_str = str(tf_path)
+        shutil.rmtree(tf_path_str)
+        sandbox_data_handler.set_tf_working_dir("")
+
+    def _create_shell_helper(self, logger: logging.Logger) -> ShellHelperObject:
+        api = CloudShellSessionContext(self._context).get_api()
+        sandbox_id = self._context.reservation.reservation_id
+        sandbox_message_service = SandboxMessagesService(api, sandbox_id, self._tf_service.name,
+                                                         self._config.write_sandbox_messages)
+        live_status_updater = LiveStatusUpdater(api, sandbox_id, self._config.update_live_status)
+        default_tags = TagsManager(self._context.reservation)
+        attr_handler = ServiceAttrHandler(self._tf_service)
+
+        return ShellHelperObject(api, sandbox_id, self._tf_service, logger, sandbox_message_service,
+                                 live_status_updater, attr_handler, default_tags)
+
+    def _create_tf_service(self) -> TerraformServiceObject:
+        api = CloudShellSessionContext(self._context).get_api()
+        reservation_id = self._context.reservation.reservation_id
+        cloudshell_model_name = self._context.resource.model
+        name = self._context.resource.name
+
+        return TerraformServiceObject(api, reservation_id, name, cloudshell_model_name)
+
+    def _does_working_dir_exists(self, dir: str) -> bool:
+        return dir and os.path.isdir(dir)
+
+    def _create_tf_proc_executer(self, sandbox_data_handler, shell_helper, tf_working_dir):
+        backend_handler = BackendHandler(shell_helper, tf_working_dir, sandbox_data_handler.get_tf_uuid())
+        input_output_service = InputOutputService(shell_helper, self._config.inputs_map, self._config.outputs_map)
+        tf_proc_executer = TfProcExec(shell_helper, sandbox_data_handler, backend_handler, input_output_service)
+        return tf_proc_executer
```

Comment:
in a previous commit it was in a factory, why did you move it back?

Reply:
unexplainable force...